22-4

KinCadeFoster opened the pull request:
- не понял что в итоге нужно жделать было) накидал что то, но уверен что не то)
- прощу разъяснить, что в кеонечном итоге необходимо сделать =)

KinCadeFoster committed:
22-4
- не понял что в итоге нужно жделать было) накидал что то, но уверен что не то)
- прощу разъяснить, что в кеонечном итоге необходимо сделать =)

ievetrovdev commented on src/main/kotlin/lesson_22/lesson22_task4.kt:
```diff
@@ -0,0 +1,55 @@
+    }
+}
+
+fun main() {
```
Решение не соответствует условию задачи. В задаче требуется создать класс MainScreenViewModel и в нем симулировать процесс загрузки данных с сервера, а также создать data class MainScreenState для хранения состояния. В решении отсутствует класс MainScreenViewModel, а метод loadData() реализован некорректно.

ievetrovdev reviewed (changes requested):
В этой задаче необходимо создать класс MainScreenViewModel, который будет отвечать за управление состоянием первого экрана приложения. В классе должен быть определен внутренний дата класс MainScreenState, который будет хранить текущее состояние экрана. Этот класс должен содержать два свойства: data (загружаемые данные) и isLoading (индикатор процесса загрузки).

KinCadeFoster committed:
22-4
- Добавил класс MainScreenViewModel
- поправил логику
- добавил getStatus() для получения состояния

ievetrovdev commented on src/main/kotlin/lesson_22/lesson22_task4.kt:
```diff
@@ -0,0 +1,31 @@
+    )
+    private var state = MainScreenState()
+
+    fun getStatus() {
```
Метод getStatus() не должен выводить информацию в консоль, а возвращать текущее состояние. Это позволит использовать этот метод для отображения состояния во View.

ievetrovdev commented on src/main/kotlin/lesson_22/lesson22_task4.kt:
```diff
@@ -0,0 +1,31 @@
+        else println("Все в полрядке, данные загружены")
+    }
+
+    fun loadData(){
```
В методе loadData() необходимо добавить задержку, имитирующую процесс загрузки данных с сервера.

ievetrovdev commented on src/main/kotlin/lesson_22/lesson22_task4.kt:
```diff
@@ -0,0 +1,31 @@
+}
+
+
+fun main() {
```
В методе main() необходимо вызвать метод getStatus() после загрузки данных, чтобы проверить, что состояние изменилось

KinCadeFoster committed:
22-4
- getStatus() - возвращает текущее состояние
- добавлена задержка loadData()
- main() уже вызывался метод после загрузки данных getStatus()

ievetrovdev reviewed (approved):
Теперь все верно!
